### DISCUSSION TITLE
reverse animation

### DISCUSSION THREAD

**Discussion by avinashgpai** [google/model-viewer#2231] [Q&A]
### Description

Well I just wanted to know if it is possible to reverse the animation? 

I have created button named next_clip that plays the animation in forward.
I wanted to create another button prev_clip that plays the animation in backward.

**Comment by elalish** [comment]
We don't currently expose an animation-rate attribute, though we could. That would allow negative values in order to play backwards. Is that a feature you'd like to see?

In the meanwhile, you can do this manually by calling `pause()` and then setting `currentTime` in a `requestAnimationFrame()` callback.

**Comment by avinashgpai** [comment]
Thanks this is what I wanted for now, subtracting from currentTime will sets current time in backwards but I would love to see animation playing in reverse. Currently I was able to set current time backwards and play from that updated time.

Thanks @elalish

**Comment by canberka** [comment]
Any update for reverse animation?? and i reallt want to see `pause()` and relative codes documentation if it is possible

**Comment by elalish** [comment]
@canberka It's all here: https://modelviewer.dev/docs/#animation-properties

**Comment by canberka** [comment]
Hello again @avinashgpai @elalish,

i cant manage to reverse animation... Can you help me for that  if it is possible? I would really appreciate it

![image](https://user-images.githubusercontent.com/36857390/112846539-d5c37d80-90ae-11eb-89e1-347aa458fa7c.png)

there is 2 button which are forward and backward...
but i confused to add this requestanimationframe()

**Comment by elalish** [comment]
You're going to need to read up on `requestAnimationFrame()` on MDN. It's a modern replacement for `setInterval`.

**Comment by sercandemirtas** [comment]
Hey @canberka I tried and update your codes and it is like;
 
https://jsfiddle.net/aqfrut56/1/


and demo link for this project; 

https://relaxed-neumann-b6600b.netlify.app/test.html

The problem is when we click first button it is going up and this is what we are trying to do but when we click second button it is going down suddenly. I cant manage to setting up the current time for second button (reverse animation). I dont have enough information about "It's a modern replacement for setInterval" which @elalish suggest to you. Maybe my codes makes idea for you but I think it should be more easy.

**Comment by qiaoxide** [comment]
Writing in this way can achieve forward and reverse animation
![image](https://user-images.githubusercontent.com/12377489/222220685-3658f007-1c64-4561-b5b0-a11f340b2d58.png)
![image](https://user-images.githubusercontent.com/12377489/222220376-bc9346ba-372a-4d89-b504-9fcebbe8acdb.png)
![image](https://user-images.githubusercontent.com/12377489/222220199-401c650e-219a-45d3-84f6-b0d867b3050a.png)
